tools/clean-ecr

```diff
@@ -0,0 +1,150 @@
+#! /usr/bin/env python
+
+import os
+import sys
+import subprocess
+import argparse
+import json
+
+"""
+This tool scans an ECR repository to find images with vulnerabilities.
+Images that are tagged are ignored.  All untagged images that are found to have
+vulnerabilities are considered to be candidates for deletion.
+
+Interactive and automated actions are both supported.  By default, this tool
+will ask you to confirm deletion of each image that has been found to be
+insecure.
+
+Run "clean-ecr --help" for usage information.
+"""
+
+ADMIN_ARN = os.environ.get('ADMIN_ARN')
+ECR_ACCOUNT_TO_USE = os.environ.get('ECR_ACCOUNT_TO_USE')
+IMAGE_REPO = os.environ.get('IMAGE_REPO')
+IMAGE_UBUNTU_NAME = os.environ.get('IMAGE_UBUNTU_NAME')
+
+CANDIDATES_FOR_DELETION = []
+
+DELETE_LEVELS = {
+    "CRITICAL" : ["CRITICAL"],
+    "HIGH": ["CRITICAL", "HIGH"],
+    "MEDIUM" : ["CRITICAL", "HIGH", "MEDIUM"],
+    "LOW" : ["CRITICAL", "HIGH", "MEDIUM", "LOW"],
+    "INFORMATIONAL" : ["CRITICAL", "HIGH", "MEDIUM", "LOW", "INFORMATIONAL"],
+    "ALL" : ["CRITICAL", "HIGH", "MEDIUM", "LOW", "INFORMATIONAL"],
+}
+
+
+class Image(object):
+
+    def __init__(self, digest):
+        self.digest = digest
+        self.insecure = True
+        self.can_delete = False
+        self.to_delete = False
+
+    def scan(self, os_version, min_severity):
+        scan_output = None
+        try:
+            proc = subprocess.Popen(
+                f'image-scan-report {os_version} {min_severity} --image-digest {self.digest}'.split(),
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE
+            )
+            scan_output, err = proc.communicate()
+            if proc.returncode == 0:
+                self.insecure = False
+        except Exception as e:
+            raise e
+
+        if self.insecure:
+            print()
+            # THIS IS EMPTY
+            print(scan_output)
```

Comment:
@jaytmiller @jmatuskey - can either of you spot why this output may be empty?  `image-scan-report` should always produce an output.

[I had realized that I was flagging everything as "deletable" if the output was non-0, but I was checking against the specified error level, so I am trying to actually read in the output to parse through it.]